internetarchive/utils.py

```diff
@@ -78,24 +80,24 @@ def validate_s3_identifier(string):
     return True
 
 
-def needs_quote(s):
+def needs_quote(s: str) -> bool:
     try:
         s.encode('ascii')
     except (UnicodeDecodeError, UnicodeEncodeError):
         return True
     return re.search(r'\s', s) is not None
 
 
-def norm_filepath(fp):
+def norm_filepath(fp: bytes | str) -> str:
```

Comment:
Not for you to fix here, just as a general remark: We should change this due to having dropped the Python 2 support.